processing-service-core/api/controllers/globalConfig.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var globalConfigModel = require('../models/globalConfig.js');
+var logger = require('../helpers/loggingHelper.js');
+var httpHelper = require('../helpers/httpPromiseHelper.js');
+
+module.exports = {
+  createGlobalConfig: createGlobalConfig,
+  deleteGlobalConfig: deleteGlobalConfig,
+  getGlobalConfig: getGlobalConfig,
+  getGlobalConfigs: getGlobalConfigs
+};
+
+function createGlobalConfig(req, res, next) {
+  var globalConfigParams = {
+    key: req.swagger.params['key'].value,
+    default: req.swagger.params['default'].value,
+    description: req.swagger.params['description'].value
+  };
+  logger.info('Creating global configuration using parameters: '+ JSON.stringify(globalConfigParams));
+  var createPromise = globalConfigModel.createGlobalConfig(globalConfigParams);
+  httpHelper.writeCreatePromiseJSONResultToResponse(createPromise, res);
```

Comment:
This should be using the update promise method as with tenant configs [here](https://github.com/CAFDataProcessing/data-processing-service/blob/SCMOD-3111/processing-service-core/api/controllers/tenant.js#L51) because this can be either a create or an update. The method name should be changed to 'setGlobalConfig' to reflect that this can be either update or delete.